docs/proposals/version_documentation.md

```diff
@@ -0,0 +1,162 @@
+
+```|- website
+    |- archetypes
+    |- layout
+    |- data
+    |- static
+    |- resources
+    |- tmp
+        |- public
+        |- docs-pre-processed
+```
+We want to add aditional `versioned` folder within the website's `tmp` directory. For example:
+
+```|- website
+    |- archetypes
+    |- layout
+    |- data
+    |- static
+    |- resources
+    |- tmp
+        |- public
+        |- docs-pre-processed
+        |- versioned
+            |- master
+            |- version 0.13.0
+            |- version 0.12.2
+            |- other-folder-for-other-releases
+```
+_NOTE: `tmp` directory is not committed, just temporarily built. The current version of docs lives in the `master` folder_
+
+
+#### 3. Building a versioning plugin
+
+Creating a plugin that can automate these processes would save us alot of development time and stress. This approach promises to be very useful when it comes to versioning different release in the Thanos website.
+
+
+##### Workflow
+
+1. Developer makes all the necessary edit on `/docs` on master
+2. Developer proceeds by committing a new Release (i.e Release 0.x)
+3. CI run some `make web` or `Make web-serve`
+4. Before anything, CI generate those docs and stores in a `versioned-tmp` folder
+5. rest of web command is done
+
+_NOTE: generated docs are not committed, just temporarily built._
+
+## FAQ
+
+This section consists of some important questions and answers that are frequently asked by users.
+
+##### How to specify what part of docs you want to version without cloning every part of the docs? docs.yaml
+
+We hope to have a single flexible configuration file (`docs.yaml`) that will help the developer in specifying what part of docs he/she needs without cloning all the docs. We could have this as a single file on master. The config file will look like this:
+
+```versioned:
+   default:
+   - docs/components/*.md
+   - docs/storage.md
+
+  overrides:
+     release-0.12:
+     - docs/storage2221241.md
+```
+##### Alternative path
+
+We could have this file on master, so the current `make web` will
+
+1. Parse on master `docs.yaml` and decide.
+1. Go to that particular release (`i.e release-0.10`)
+1. Is there docs.yaml?
+    * Yes - parse and decide
+    * No - use docs.yaml from master
+
+The design of docs.yaml will look like this:
+
+```versioned:
+   - docs/components/*.md
+   - docs/storage2221241.md
+```
+##### How would the plugin look like?
+
+CLI (`make generate-versioned-docs`)
+
+##### Workplan in terms of plugin placement?
+
+We hope to start in Thanos project, then move project outside to a separate repo.
+
+##### How we do fixes to release docs for older release e.g v0.12?
+
+We will edit the particular release (release-0.12) and commit. Then the tool fetches the latest on this branch and use it to generate docs. We expect docs for minor version is what matters - not breaking with patch releases.
```

Comment:
```suggestion
We will edit the particular release (release-0.12) and commit. Then the tool fetches the latest on this branch and uses it to generate docs. We expect docs for minor version is what matters - not breaking with patch releases.
```

Don't follow `We expect docs for minor version is what matters - not breaking with patch releases.` sentence, can you rephrase?

Reply:
Alright What about this: `We expect to fetch docs for minor releases without breaking these releases with patches. We encourage immutability across all our releases in Thanos. `